lib/graphql/list_type.rb

```diff
@@ -17,8 +17,13 @@ def to_s
     "[#{of_type.to_s}]"
   end
 
-  def coerce_input(value)
-    inner_type = of_type
-    value.map { |item| inner_type.coerce_input!(item) }
+  def valid_non_null_input?(value)
+    value = [value] unless value.is_a?(Array)
```

Comment:
Sure, I'll split it out into a separate PR, since this only fixes it for variables. Literals would still need to be fixed because of the check you linked to.